ground_truth_slam/plugin.cpp

```diff
@@ -26,29 +27,51 @@ class ground_truth_slam : public plugin {
 	}
 
 	void feed_ground_truth(const imu_cam_type *datum) {
-		ullong rounded_time = datum->dataset_time;
+		ullong rounded_time = floor(datum->dataset_time / 10000);
 		_m_sensor_data_it = _m_sensor_data.find(rounded_time);
 
 		if (_m_sensor_data_it == _m_sensor_data.end()) {
-			#ifndef NDEBUG
+#ifndef NDEBUG
 				std::cout << "True pose not found at timestamp: " << rounded_time << std::endl;
-			#endif
+#endif
 			return;
 		}
 
 		pose_type* true_pose = new pose_type{_m_sensor_data_it->second};
 		true_pose->sensor_time = datum->time;
-		// std::cout << "The pose was found at " << true_pose->position[0] << ", " << true_pose->position[1] << ", " << true_pose->position[2] << std::endl; 
-
+		// Ground truth position offset is the first ground truth position
+		if (first_time) {
+			first_time = false;
+			_m_ground_truth_offset->put(new Eigen::Vector3f{true_pose->position});
+		}
 		_m_true_pose->put(true_pose);
+
+#ifndef NDEBUG
+		std::cout << "Ground truth pose was found at T: " << rounded_time
+				  << " | "
+				  << "Pos: ("
+				  << true_pose->position[0] << ", "
+				  << true_pose->position[1] << ", "
+				  << true_pose->position[2] << ")"
+				  << " | "
+				  << "Quat: ("
+				  << true_pose->orientation.w() << ", "
+				  << true_pose->orientation.x() << ", "
+				  << true_pose->orientation.y() << ","
+				  << true_pose->orientation.z() << ")"
+				  << std::endl;
+#endif
 	}
 
 private:
 	const std::shared_ptr<switchboard> sb;
 	std::unique_ptr<writer<pose_type>> _m_true_pose;
+	std::unique_ptr<writer<Eigen::Vector3f>> _m_ground_truth_offset;
 
 	const std::map<ullong, sensor_types> _m_sensor_data;
 	std::map<ullong, sensor_types>::const_iterator _m_sensor_data_it;
+
+	bool first_time{true};
```

Comment:
Should this be set in the constructor?

Reply:
That's what I would have done but the other plugins do it here, so kept it here for "consistency".

Reply:
Gotcha. If you don't mind, lets clean as we go, since handling #45 likely won't catch issues like this.